streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -107,18 +84,23 @@ public void addChild(final ProcessorNode<?, ?> child) {
     }
 
     public void init(final InternalProcessorContext context) {
-        this.context = context;
         try {
             nodeMetrics = new NodeMetrics(context.metrics(), name, context);
-            runAndMeasureLatency(time, initDelegate, nodeMetrics.nodeCreationSensor);
+            final long startNs = time.nanoseconds();
+            if (processor != null) {
+                processor.init(context);
+            }
+            nodeMetrics.nodeCreationSensor.record(time.nanoseconds() - startNs);
```

Comment:
These changes are a bit tangential, but while looking at the metrics, I realized we were jumping through quite a few hoops to do something simple, so I simplified these latency measurements.

Reply:
I kinda see you point, however, I am wondering if there would be any runtime difference? Should the compiler be smart enough to inline the code and avoid the method calls?

Reply:
I'm not sure; it's not just a method call, but also a call to an anonymous Runnable instance we created. And one of those referenced (key/value) fields on ProcessorNode (which we put there just to pass to the Runnable). 

The runtime performance is certainly not slower, but I'd have to profile it to see if it's faster. I'm interested in doing this not so much for the performance gain (which I'd hope is minimal) as for the readability/simplicity, which is significantly improved.